### PR TITLE
[flang] Support -mabi=vec-extabi and -mabi=vec-default on AIX

### PR DESCRIPTION
**Pull request by kkwli**
This option is to enable the AIX extended and default vector ABIs.

### CODE REVIEW

File: clang/lib/Driver/ToolChains/Flang.cpp [hunk 1]
```diff
@@ -203,6 +203,30 @@ void Flang::AddAArch64TargetArgs(const ArgList &Args,
   }
 }
 
+void Flang::AddPPCTargetArgs(const ArgList &Args,
+                             ArgStringList &CmdArgs) const {
+  bool VecExtabi = false;
+  for (const Arg *A : Args.filtered(options::OPT_mabi_EQ)) {
```
Comment (tblah) [inline clang/lib/Driver/ToolChains/Flang.cpp:209]: Could the loop be simplified to something like this?
```
Arg *A = Args.getLastArg(options::OPT_mabi_EQ);
if (A) {
  StringRef V = A->getValue();
  if (V == "vec-default") {
 [...]
}
```

Comment (kkwli) [reply clang/lib/Driver/ToolChains/Flang.cpp:209]: Will do

File: flang/test/Driver/mabi.f90 [hunk 2]
```diff
@@ -0,0 +1,15 @@
+! RUN: not %flang -### -c --target=powerpc64le-unknown-linux -mabi=vec-extabi %s 2>&1 | FileCheck --check-prefix=INVALID1 %s
+! RUN: not %flang -### -c --target=x86_64-unknown-linux -mabi=vec-extabi %s 2>&1 | FileCheck --check-prefix=INVALID2 %s
+! RUN: not %flang -### -c --target=powerpc-unknown-aix -mabi=abc %s 2>&1 | FileCheck --check-prefix=INVALID3 %s
+! RUN: %flang -### -c -target powerpc-unknown-aix %s 2>&1 | FileCheck --implicit-check-not=vec-extabi %s
+! RUN: %flang -### -c -target powerpc-unknown-aix -mabi=vec-default %s 2>&1 | FileCheck --implicit-check-not=vec-extabi %s
+! RUN: %flang -### -c -target powerpc-unknown-aix -mabi=vec-extabi %s 2>&1 | FileCheck --check-prefix=EXTABI %s
+
```
Comment (jeanPerier) [inline flang/test/Driver/mabi.f90:7]: I think `! REQUIRES: target=powerpc{{.*}}` may be needed.

File: clang/lib/Driver/ToolChains/Flang.cpp [hunk 3]
```diff
@@ -203,6 +203,33 @@ void Flang::AddAArch64TargetArgs(const ArgList &Args,
...
 }
 
+void Flang::AddPPCTargetArgs(const ArgList &Args,
+                             ArgStringList &CmdArgs) const {
+  const Driver &D = getToolChain().getDriver();
+  bool VecExtabi = false;
+
+  if (const Arg *A = Args.getLastArg(options::OPT_mabi_EQ)) {
+    StringRef V = A->getValue();
+    if (V == "vec-extabi") {
```
Comment (jeanPerier) [inline clang/lib/Driver/ToolChains/Flang.cpp:213]: nit: not braces on single statement if/else.

**Review by jeanPerier — COMMENTED** [review]
Other than the "REQUIRES" in the test, LGTM.

File: clang/lib/Driver/ToolChains/Flang.cpp (hunk 3 above)
Comment (kkwli) [reply clang/lib/Driver/ToolChains/Flang.cpp:213]: Will remove the braces

File: flang/test/Driver/mabi.f90 (hunk 2 above)
Comment (kkwli) [reply flang/test/Driver/mabi.f90:7]: Sure.

**Review by DanielCChen — APPROVED** [review]
LGTM.

**Review by tblah — APPROVED** [review]
LGTM thanks

**Pull request merged**